tests/lxd/test_lxd_mount_handler.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "mock_lxd_server_responses.h"
+#include "mock_network_access_manager.h"
+#include "tests/mock_file_ops.h"
+#include "tests/mock_logger.h"
+#include "tests/stub_ssh_key_provider.h"
+#include "tests/stub_status_monitor.h"
+
+#include "src/platform/backends/lxd/lxd_mount_handler.h"
+#include "src/platform/backends/lxd/lxd_virtual_machine.h"
+
+#include <multipass/utils.h>
+#include <multipass/virtual_machine_description.h>
+#include <multipass/vm_mount.h>
+
+namespace mp = multipass;
+namespace mpt = multipass::test;
+
+namespace
+{
+class MockLXDVirtualMachine : public mp::LXDVirtualMachine
```

Comment:
Yep, will do. However, I would say one small advantage of the current inheritance is that the MOCK_METHOD list in the MockLXDVirtualMachine shows exactly what he needs, it is just `current_state()` function in this case. In the uniform MockVirtualMachineT case, it mocked a lot of things for different client classes even though some are not needed by some clients.

Reply:
I see that rather as a problem though. If the tests assume that is the only thing that needs to be mocked, they are relying on the tested code to be correct in that way, i.e., that it doesn't use anything else. Otherwise, tests run production code. 

IOW, tests should work safely against wrong code too, including code calling other methods. If the methods are mocked, that is caught and dealt with according on the personality of the mock (strict,naggy,nice) and production code is not run.